### ISSUE TITLE
Scan fails when dockle is called inside from a shell script

### ISSUE THREAD

**Issue by KEINOS** [goodwithtech/dockle#15]
**TL;DR**

On macOS, **when `dockle` command is called inside a shell script, the scan fails** as below, but running the same command directly in the terminal `dockle` analyzes perfectly.

```
FATAL	failed scan <my_image>:<my_tag>: %!w(<nil>):
```

**Description**

I have a simple shell script such as below, which builds the docker image then calls `dockle` to analyze it. (The Docker container simply [echoes "Hello World"](https://github.com/KEINOS/Sample_Dockle/blob/9049f63fb5c8f61e192c543c9938c3680cc780fe/src/hello-world.sh))

```sh
#!/bin/sh
# - Script Name: build.sh

export DOCKER_CONTENT_TRUST=1 && \
docker build --no-cache -t my_image:my_test . && \
dockle --debug --clear-cache my_image:my_test
```

The above shell script fails in the last line which `dockle` is called. But running the last command directly like below, it works.

```
$ dockle --debug --clear-cache my_image:my_test
```

**What did you expect to happen?**

- Same result of running directly in the shell.

**What happened instead?**

- Fails to scan image if `dockle` is called inside the shell script.

**Output of run with `-debug`:**

```shellsession
2019-06-13T12:05:39.979+0900	DEBUG	Fetch latest version from github
2019-06-13T12:05:41.007+0900	WARN	A new version v0.1.1 is now available! You have v0.0.18.
2019-06-13T12:05:41.007+0900	INFO	Removing image caches...
2019-06-13T12:05:41.008+0900	DEBUG	Start assessments...
2019-06-13T12:05:41.233+0900	DEBUG	Start scan : password files
2019-06-13T12:05:41.233+0900	DEBUG	Start scan : /etc/passwd
2019-06-13T12:05:41.233+0900	DEBUG	Start scan : /etc/group
2019-06-13T12:05:41.233+0900	DEBUG	Start scan : /etc/hosts
2019-06-13T12:05:41.233+0900	DEBUG	Start scan : credential files
2019-06-13T12:05:41.233+0900	DEBUG	Scan start : config file
2019-06-13T12:05:41.233+0900	DEBUG	Scan start : DOCKER_CONTENT_TRUST
2019-06-13T12:05:41.234+0900	FATAL	failed scan my_image:my_test: %!w(<nil>):
    github.com/goodwithtech/dockle/pkg/scanner.ScanImage
        /go/src/github.com/goodwithtech/dockle/pkg/scanner/scan.go:53
```

**Output of `guard -v`:**

- n/a

**How to produce the error:**

- I made a repo to produce the error.
    - [Sample_Dockle](https://github.com/KEINOS/Sample_Dockle/tree/9049f63fb5c8f61e192c543c9938c3680cc780fe) @ GitHub

**Additional details (base image name, container registry info...):**

It seems that `assessor.GetAssessments()` is not returning the `assessments` right when the `image_name` was given in the shell script.

<details><summary>ENV INFO</summary><div><br>

- Dockle Version: v0.0.18
- Host OS: macOS Mojave (OSX 10.14.5, darwin/amd64)
- `/bin/sh --version`:  GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin18)
- Docker Engine: v18.09.0 (Community Version)
- Base Image: alpine:latest
- [Dockerfile](https://github.com/KEINOS/Sample_Dockle/blob/9049f63fb5c8f61e192c543c9938c3680cc780fe/Dockerfile)

</div></details>

<details><summary>RESULT DETAILS</summary><div><br>

- Results of **running via shell script**. ([See source of build.sh](https://github.com/KEINOS/Sample_Dockle/blob/9049f63fb5c8f61e192c543c9938c3680cc780fe/build.sh))

```shellsession
$ ./build.sh
Sending build context to Docker daemon   76.8kB
Step 1/6 : FROM alpine@sha256:769fddc7cc2f0a1c35abb2f91432e8beecf83916c421420e6a6da9f8975464b6
 ---> 055936d39205
Step 2/6 : COPY src /app
 ---> 886b6c29e463
Step 3/6 : RUN addgroup appgroup &&     adduser -S dockle appgroup
 ---> Running in edea519fb364
Removing intermediate container edea519fb364
 ---> 1f08283aea47
Step 4/6 : HEALTHCHECK --interval=1h --timeout=15s --start-period=5s --retries=2 CMD [ "/app/healthcheck.sh" ]
 ---> Running in 584193c4ada6
Removing intermediate container 584193c4ada6
 ---> 76975fea7a9a
Step 5/6 : CMD [ "/app/hello-world.sh" ]
 ---> Running in d6ee59afa44a
Removing intermediate container d6ee59afa44a
 ---> 2c75e3edfb9e
Step 6/6 : USER dockle
 ---> Running in bdcb1d4f926b
Removing intermediate container bdcb1d4f926b
 ---> 4481ee34427e
Successfully built 4481ee34427e
Successfully tagged my_image:my_test
Tagging alpine@sha256:769fddc7cc2f0a1c35abb2f91432e8beecf83916c421420e6a6da9f8975464b6 as alpine:latest
2019-06-13T12:34:07.573+0900	DEBUG	Fetch latest version from github
2019-06-13T12:34:08.455+0900	WARN	A new version v0.1.1 is now available! You have v0.0.18.
2019-06-13T12:34:08.455+0900	INFO	Removing image caches...
2019-06-13T12:34:08.456+0900	DEBUG	Start assessments...
2019-06-13T12:34:08.686+0900	DEBUG	Start scan : password files
2019-06-13T12:34:08.686+0900	DEBUG	Start scan : /etc/passwd
2019-06-13T12:34:08.686+0900	DEBUG	Start scan : /etc/group
2019-06-13T12:34:08.686+0900	DEBUG	Start scan : /etc/hosts
2019-06-13T12:34:08.686+0900	DEBUG	Start scan : credential files
2019-06-13T12:34:08.686+0900	DEBUG	Scan start : config file
2019-06-13T12:34:08.687+0900	DEBUG	Scan start : DOCKER_CONTENT_TRUST
2019-06-13T12:34:08.687+0900	FATAL	failed scan my_image:my_test: %!w(<nil>):
    github.com/goodwithtech/dockle/pkg/scanner.ScanImage
        /go/src/github.com/goodwithtech/dockle/pkg/scanner/scan.go:53
$ 
$ echo $?
1
```

- Results **running via shell command**

```shellsession
$ # Same command in the shell script
$ dockle --debug --clear-cache my_image:my_test
2019-06-13T12:45:35.765+0900	DEBUG	Fetch latest version from github
2019-06-13T12:45:36.792+0900	WARN	A new version v0.1.1 is now available! You have v0.0.18.
2019-06-13T12:45:36.792+0900	INFO	Removing image caches...
2019-06-13T12:45:36.793+0900	DEBUG	Start assessments...
2019-06-13T12:45:37.043+0900	DEBUG	Start scan : password files
2019-06-13T12:45:37.043+0900	DEBUG	Start scan : /etc/passwd
2019-06-13T12:45:37.043+0900	DEBUG	Start scan : /etc/group
2019-06-13T12:45:37.043+0900	DEBUG	Start scan : /etc/hosts
2019-06-13T12:45:37.043+0900	DEBUG	Start scan : credential files
2019-06-13T12:45:37.043+0900	DEBUG	Scan start : config file
2019-06-13T12:45:37.043+0900	DEBUG	Scan start : DOCKER_CONTENT_TRUST
2019-06-13T12:45:37.043+0900	DEBUG	End assessments...
2019-06-13T12:45:37.043+0900	DEBUG	There is no .dockleignore file
PASS	- CIS-DI-0001: Create a user for the container
WARN	- CIS-DI-0005: Enable Content trust for Docker
	* export DOCKER_CONTENT_TRUST=1 before docker pull/build
PASS	- CIS-DI-0006: Add HEALTHCHECK instruction to the container image
PASS	- CIS-DI-0007: Do not use update instructions alone in the Dockerfile
PASS	- CIS-DI-0008: Remove setuid and setgid permissions in the images
PASS	- CIS-DI-0009: Use COPY instead of ADD in Dockerfile
PASS	- CIS-DI-0010: Do not store secrets in ENVIRONMENT variables
PASS	- CIS-DI-0010: Do not store secret files
PASS	- DKL-DI-0001: Avoid sudo command
PASS	- DKL-DI-0002: Avoid sensitive directory mounting
PASS	- DKL-DI-0003: Avoid apt-get/apk/dist-upgrade
PASS	- DKL-DI-0004: Use apk add with --no-cache
PASS	- DKL-DI-0005: Clear apt-get caches
PASS	- DKL-DI-0006: Avoid latest tag
PASS	- DKL-LI-0001: Avoid empty password
PASS	- DKL-LI-0002: Be unique UID
PASS	- DKL-LI-0002: Be unique GROUP
$
$ echo $?
0
```

</div></details>

**Comment by KEINOS** [comment]
I updated `dockle` version `v0.0.18` -> `v0.1.1` but still gives me the same error.

**Comment by tomoyamachi** [comment]
@KEINOS 
Thank you for report!!
I fixed a terrible bug.

Please update v0.1.3.

**Comment by KEINOS** [comment]
> I fixed a terrible bug.
> Please update v0.1.3.

Thank you! Now it works like a charm! :tada:

<details><summary>Issue Closed!</summary><div><br>

```shellsession
$ dockle --version
dockle version 0.1.5
$
$ ./build.sh
- Building: Docker image ...
Sending build context to Docker daemon   76.8kB
Step 1/6 : FROM alpine@sha256:769fddc7cc2f0a1c35abb2f91432e8beecf83916c421420e6a6da9f8975464b6
 ---> 055936d39205
Step 2/6 : COPY src /app
 ---> 2fbd025a8090
Step 3/6 : RUN addgroup appgroup &&     adduser -S dockle appgroup
 ---> Running in 7c0d77f0755d
Removing intermediate container 7c0d77f0755d
 ---> 71335af5a605
Step 4/6 : HEALTHCHECK --interval=1h --timeout=15s --start-period=5s --retries=2 CMD [ "/app/healthcheck.sh" ]
 ---> Running in 644188b31850
Removing intermediate container 644188b31850
 ---> f8a2ac5b2711
Step 5/6 : CMD [ "/app/hello-world.sh" ]
 ---> Running in 2a2e45ae20ef
Removing intermediate container 2a2e45ae20ef
 ---> f75fc6e8f274
Step 6/6 : USER dockle
 ---> Running in 76679dbb8881
Removing intermediate container 76679dbb8881
 ---> f9db4a614ea2
Successfully built f9db4a614ea2
Successfully tagged my_image:my_test
Tagging alpine@sha256:769fddc7cc2f0a1c35abb2f91432e8beecf83916c421420e6a6da9f8975464b6 as alpine:latest
- Auditing Security: Docker image ...
2019-06-14T02:03:14.334+0900	DEBUG	Fetch latest version from github
2019-06-14T02:03:16.007+0900	INFO	Removing image caches...
2019-06-14T02:03:16.008+0900	DEBUG	Start assessments...
2019-06-14T02:03:16.272+0900	DEBUG	Start scan : password files
2019-06-14T02:03:16.272+0900	DEBUG	Start scan : /etc/passwd
2019-06-14T02:03:16.272+0900	DEBUG	Start scan : /etc/group
2019-06-14T02:03:16.272+0900	DEBUG	Start scan : /etc/hosts
2019-06-14T02:03:16.272+0900	DEBUG	Start scan : credential files
2019-06-14T02:03:16.272+0900	DEBUG	Scan start : config file
2019-06-14T02:03:16.273+0900	DEBUG	Scan start : DOCKER_CONTENT_TRUST
2019-06-14T02:03:16.273+0900	DEBUG	End assessments...
2019-06-14T02:03:16.273+0900	DEBUG	There is no .dockleignore file
PASS	- CIS-DI-0001: Create a user for the container
PASS	- CIS-DI-0005: Enable Content trust for Docker
PASS	- CIS-DI-0006: Add HEALTHCHECK instruction to the container image
PASS	- CIS-DI-0007: Do not use update instructions alone in the Dockerfile
PASS	- CIS-DI-0008: Remove setuid and setgid permissions in the images
PASS	- CIS-DI-0009: Use COPY instead of ADD in Dockerfile
PASS	- CIS-DI-0010: Do not store secrets in ENVIRONMENT variables
PASS	- CIS-DI-0010: Do not store secret files
PASS	- DKL-DI-0001: Avoid sudo command
PASS	- DKL-DI-0002: Avoid sensitive directory mounting
PASS	- DKL-DI-0003: Avoid apt-get/apk/dist-upgrade
PASS	- DKL-DI-0004: Use apk add with --no-cache
PASS	- DKL-DI-0005: Clear apt-get caches
PASS	- DKL-DI-0006: Avoid latest tag
PASS	- DKL-LI-0001: Avoid empty password
PASS	- DKL-LI-0002: Be unique UID
PASS	- DKL-LI-0002: Be unique GROUP
```

</div></details>
